src/main/java/org/eclipse/microprofile/starter/log/DynamoDBLogger.java

```diff
@@ -0,0 +1,318 @@
+                if (Log.isLoggable(Level.FINE)) {
+                    try (BufferedReader reader = new BufferedReader(new InputStreamReader(myURLConnection.getErrorStream()))) {
+                        String line;
+                        while ((line = reader.readLine()) != null) {
+                            Log.log(Level.FINE, line);
+                        }
+                    }
+                }
+                return Status.NOK;
+            }
+        } catch (IOException e) {
+            Log.log(Level.SEVERE, "HttpURLConnection failed while reading error stream.", e);
+            return Status.NOK;
+        }
+
+        if (Log.isLoggable(Level.FINE)) {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(myURLConnection.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    Log.log(Level.FINE, line);
+                }
+            } catch (IOException e) {
+                Log.log(Level.SEVERE, "HttpURLConnection failed while reading input stream.", e);
+                return Status.NOK;
+            }
+        }
+
+        return Status.OK;
+    }
+
+    private static void validateEngineData(final EngineData engineData) {
+        if (engineData.getMpVersion() == null || engineData.getMpVersion().length() < 1) {
+            throw new IllegalArgumentException("mpVersion must not be null/empty.");
+        }
+        if (engineData.getSupportedServer() == null || engineData.getSupportedServer().length() < 1) {
+            throw new IllegalArgumentException("supportedServer must not be null/empty.");
+        }
+        if (engineData.getBeansxmlMode() == null || engineData.getBeansxmlMode().length() < 1) {
+            throw new IllegalArgumentException("beansxmlMode must not be null/empty.");
+        }
+        if (engineData.getSelectedSpecs() == null) {
+            throw new IllegalArgumentException("selectedSpecs must not be null");
+        }
+        if (engineData.getMavenData().getArtifactId() == null || engineData.getMavenData().getArtifactId().length() < 1) {
+            throw new IllegalArgumentException("artifacId must not be null/empty.");
+        }
+        if (engineData.getMavenData().getGroupId() == null || engineData.getMavenData().getGroupId().length() < 1) {
+            throw new IllegalArgumentException("groupId must not be null/empty.");
+        }
+    }
+
+    /**
+     * Expensive blocking method that creates signed request and does HTTP POST to AWS DynamoDB API
+     *
+     * @param engineData Logged as is except for groupIdArtifacId which is merely used to enhance log message hash.
+     * @return Either Status.OK or Status.NOK if the AWS DynamoDB access is configured.
+     * The caller is not expected to be able to do anything about the error anyway.
+     */
+    public Status log(final EngineData engineData) {
```

Comment:
@rdebusscher is it O.K. that the inside of the method is not synchronized? Wouldn't ApplicationScoped just hand over the same instance to multiple threads the way we use it...?

Reply:
No need to synchronize. The DynamoDBLogger class doesn't hold any property which is user or thread specific and doesn't use any class which isn't thread-safe (in fact no properties at all). 

So the class is thread-safe and thus not synchronized keyword needed.

Use of synchronized should only be in very specific cases when no alternative is possible as it is a performance killer.

Reply:
Why the Status return, it is not used?

Reply:
@rdebusscher Ad race: O.K. thx.
Ad Status: originally I though the caller would react to that, but it has no meaning now. I'll remove it tomorrow.